src/OrleansServiceBus/Providers/Streams/EventHub/EventHubBatchContainer.cs

```diff
@@ -0,0 +1,123 @@
+﻿/*
+Project Orleans Cloud Service SDK ver. 1.0
+ 
+Copyright (c) Microsoft Corporation
+ 
+All rights reserved.
+ 
+MIT License
+
+Permission is hereby granted, free of charge, to any person obtaining a copy of this software and 
+associated documentation files (the ""Software""), to deal in the Software without restriction,
+including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense,
+and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so,
+subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO
+THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS
+OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT,
+TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+*/
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.ServiceBus.Messaging;
+using Newtonsoft.Json;
+using Orleans.Runtime;
+using Orleans.Serialization;
+using Orleans.ServiceBus.Providers.Streams.EventHub;
+using Orleans.Streams;
+
+namespace OrleansServiceBusUtils.Providers.Streams.EventHub
+{
+    [Serializable]
+    internal class EventHubBatchContainer : IBatchContainer
+    {
+        [JsonProperty]
+        private readonly EventHubSequenceToken token;
+
+        [JsonProperty]
+        private readonly byte[] bytes;
+
+        public Guid StreamGuid { get; private set; }
+        public string StreamNamespace { get; private set; }
+        public StreamSequenceToken SequenceToken { get { return token; } }
+
+        private Body payload;
+        private Body Payload
```

Comment:
Maybe easier (more clear to understand what is going on) just to get rid of Body class and have 2 fields Events and RC and check if null and init if needed before every use. There are only a couple of uses. I think it will be more clear.